components/nodemanager-service/pgdb/project_update.go

```diff
@@ -285,3 +193,185 @@ func (node *NodeProjectData) getValues(valuesType string) []string {
+	'8', '9', 'a', 'b',
+	'c', 'd', 'e', 'f',
+}
+
+func uuidPrefixRanges16() []string {
+	prefixes := make([]string, 16)
+	for i := 0; i < 16; i++ {
+		prefix := make([]byte, 1)
+		prefix[0] = uuidChars[i]
+		prefixes[i] = string(uuidChars[i])
+	}
+	return prefixes
+}
+
+func (db *DB) ListProjectUpdateTasks(ctx context.Context) ([]project_update_lib.SerializedProjectUpdateTask, error) {
+
+	var count int
+	row := db.Db.QueryRow(countNodesProjectData)
+	if err := row.Scan(&count); err != nil {
+		return nil, err
+	}
+
+	if count <= 50000 {
+		tasks := make([]project_update_lib.SerializedProjectUpdateTask, 1)
+		tasks[0] = project_update_lib.SerializedProjectUpdateTask{
+			Priority: 1 << 62,
+			Params: map[string]string{
+				"s": "0",
+				"e": "g",
+			},
+		}
+		return tasks, nil
+	}
+
+	prefixes := uuidPrefixRanges16()
+	logrus.Infof("Found %d nodes, splitting into %d chunks", count, len(prefixes))
+	tasks := make([]project_update_lib.SerializedProjectUpdateTask, len(prefixes))
+	for i := 0; i < len(prefixes); i++ {
+		start := prefixes[i]
+		end := ""
+		if i+1 < len(prefixes) {
+			end = prefixes[i+1]
+		} else {
+			end = "g"
+		}
+
+		tasks[i] = project_update_lib.SerializedProjectUpdateTask{
+			Priority: 1 << 62,
+			Params: map[string]string{
+				"s": start,
+				"e": end,
+			},
+		}
+	}
+	return tasks, nil
+
+}
+
+func (db *DB) RunProjectUpdateTask(ctx context.Context, projectUpdateID string,
+	params map[string]string, projectTaggingRules map[string]*iam_v2.ProjectRules) (
+	project_update_lib.SerializedProjectUpdateTaskID,
+	project_update_lib.SerializedProjectUpdateTaskStatus,
+	error) {
+
+	start := params["s"]
+	end := params["e"]
+
+	logrus.WithFields(logrus.Fields{
+		"start": start,
+		"end":   end,
+	}).Info("Running project update task for nodemanager")
+
+	if start == "" || end == "" {
+		return "", project_update_lib.SerializedProjectUpdateTaskStatus{}, errors.New("Invalid params")
+	}
+
+	iter := nodeDataIterator{
+		rangeStart: start,
+		rangeEnd:   end,
+		db:         db.DbMap,
+	}
+
+	for {
+		nodes, err := iter.Next()
+		if err != nil {
+			if err == io.EOF {
+				logrus.WithFields(logrus.Fields{
+					"start": start,
+					"end":   end,
+				}).Info("project update task complete for nodemanager")
+				return "", project_update_lib.SerializedProjectUpdateTaskStatus{
+					State:              project_update_lib.SerializedProjectUpdateTaskSuccess,
+					PercentageComplete: 1.0,
+				}, nil
+
+			}
+			return "", project_update_lib.SerializedProjectUpdateTaskStatus{}, err
+		}
+
+		if err := db.updateNodes(nodes, projectTaggingRules); err != nil {
+			return "", project_update_lib.SerializedProjectUpdateTaskStatus{}, err
+		}
+	}
+
+}
+
+func (db *DB) MonitorProjectUpdateTask(ctx context.Context,
+	projectUpdateID string, id project_update_lib.SerializedProjectUpdateTaskID) (
+	project_update_lib.SerializedProjectUpdateTaskStatus, error) {
+
+	return project_update_lib.NoStatus, errors.New("Unimplemented")
```

Comment:
We do not need to implement it because the task is synchronous. It returns with fail or success and not an id that can later be checked like we do for ES